src/javascript/DesignSystem/DatePicker/TimeSelector/TimeSelector.jsx

```diff
@@ -31,14 +31,26 @@ const style = theme => ({
     }
 });
 
-const hours = new Array(24).fill().reduce((acc, _, i) => {
-    return [...acc, `${i}:00`, `${i}:30`];
+const hours = disabledHours => new Array(48).fill().reduce((acc, _, i) => {
```

Comment:
Some code and process could be avoided if there is no disabledHours. please add a check on disabledHours to see if it's necessary to process the disabledHours.

Reply:
done ! 
Thank you for the review